test/routes.test.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const request = require('supertest');
+const deviceRoutes = require('../src/routes');
+const { server, db } = require('./createTestServer');
+
+
+const devices = deviceRoutes(server);
+
+
+function dropTestCollection(database) {
+  database.db.dropCollection('devices', (err, result) => {
+    console.log('db dropped');
+  });
+}
+
+
+afterEach(() => dropTestCollection(db),
+);
+
+describe('POST /devices', () => {
+  describe('When given a name', () => {
+    it('returns successful API response', () => (
+      request(devices)
+        .post('/devices').send({ name: 'Arianna', id: 'test-id' })
+        .then((response) => {
+          expect(response.statusCode).toBe(200);
+          expect(response.text).toBe('{"code":200,"message":"New device added: Arianna test-id"}');
```

Comment:
2 things:
1. it is typical for a POST request to come back as a `201`.
2. the status code is its own thing, it does not need to be a part of the json body.